packages/itwin/tree-widget/src/components/trees/VisibilityTreeEventHandler.ts

```diff
@@ -101,6 +104,20 @@ export class VisibilityTreeEventHandler extends UnifiedSelectionTreeEventHandler
     return items.filter((item) => this._selectionPredicate!(item));
   }
 
+  public override async onNodeDoubleClick({ nodeId }: TreeNodeEventArgs) {
+    const model = this.modelSource.getModel();
+    const node = model.getNode(nodeId);
+
+    if(!node || !isPresentationTreeNodeItem(node.item) || ModelsVisibilityHandler.getNodeType(node.item) !== ModelsTreeNodeType.Element) {
+      return;
+    }
+
+    const instancesNodeKey = node.item.key as ECInstancesNodeKey;
```

Comment:
Use typeguard instead of casting. https://www.itwinjs.org/reference/presentation-common/hierarchies/nodekey/nodekey.isinstancesnodekey/